media_kit_video/lib/src/utils/wakelock.dart

```diff
@@ -0,0 +1,53 @@
+/// This file is a part of media_kit (https://github.com/alexmercerind/media_kit).
+///
+/// Copyright © 2021 & onwards, Julien Muret <birros@protonmail.com>.
+/// All rights reserved.
+/// Use of this source code is governed by MIT license that can be found in the LICENSE file.
+import 'package:flutter/widgets.dart';
+import 'package:wakelock_plus/wakelock_plus.dart';
+
+/// {@template wakelock}
+/// Wakelock
+/// --------
+///
+/// This class acquires & releases the wakelock based on reference count.
+///
+/// {@endtemplate}
+class Wakelock {
+  /// {@macro wakelock}
+  Wakelock();
+
+  /// Marks the wakelock as enabled for this instance.
+  void enable() {
+    if (!_enabled) {
+      _enabled = true;
+      _count++;
+      _update();
+    }
+  }
+
+  /// Marks the wakelock as disabled for this instance.
+  void disable() {
+    if (_enabled) {
+      _enabled = false;
+      _count--;
+      _update();
+    }
+  }
+
+  /// Acquires the wakelock if enabled count is greater than 0.
+  void _update() {
+    if (_count > 0) {
+      WakelockPlus.enable().catchError((_) {});
+    } else {
+      WakelockPlus.disable().catchError((_) {});
+    }
+    debugPrint('media_kit: wakelock: _count = $_count');
+  }
+
+  /// Whether the wakelock is enabled for this instance.
+  bool _enabled = false;
+
+  /// The number of [Wakelock] instances in enabled state.
+  static int _count = 0;
```

Comment:
I find this count-based implementation bit simpler to understand.

A single static field & we don't need to store instances inside a set.

Reply:
Hmm, this class is useless ^^
You can delete it, because `WakelockPlus` does the same thing, but statically...

My first implementation was based on counters, but it has side-effects, as counters often do.
If `enable` or `disable` is called more than once then the counter becomes inconsistent, for example if the video is paused before being disposed.

Reply:
I think you are probably reading it wrong, each instance can only cause an increment of 1 to the `_count`.

https://github.com/alexmercerind/media_kit/blob/c7b46bfd38c4b4311e049d82286e4b74903bee17/media_kit_video/lib/src/utils/wakelock.dart#L22
https://github.com/alexmercerind/media_kit/blob/c7b46bfd38c4b4311e049d82286e4b74903bee17/media_kit_video/lib/src/utils/wakelock.dart#L31

Reply:
Yep, you are right. Sorry. I didn’t see the static counter…